### ISSUE TITLE
Adding better line editing in interactive mode

### ISSUE THREAD

**Issue by Slider2k** [ggerganov/llama.cpp#4524]
# Prerequisites

Please answer the following questions for yourself before submitting an issue.

- [x] I am running the latest code. Development is very rapid so there are no tagged versions as of now.
- [x] I carefully followed the [README.md](https://github.com/ggerganov/llama.cpp/blob/master/README.md).
- [x] I [searched using keywords relevant to my issue](https://docs.github.com/en/issues/tracking-your-work-with-issues/filtering-and-searching-issues-and-pull-requests) to make sure that I am creating a new issue that is not already open (or closed).
- [x] I reviewed the [Discussions](https://github.com/ggerganov/llama.cpp/discussions), and have a new bug or useful enhancement to share.

# Feature Description

Ability to move cursor to edit and fix the prompt while using interactive mode.

# Motivation

Using llama.cpp in interactive mode I (and I bet other users too) can occasionally make typos writing prompts that I notice only after I finish writing. Unfortunately it's not possible currently to move the cursor to the error and fix it, you have to delete everything after the error, fix it and then retype everything you have deleted. Having better line editing facilities in interactive mode would be a great QoL improvement.
